src/SignalR/common/Shared/ReflectionHelper.cs

```diff
@@ -59,4 +66,7 @@ public static bool IsIAsyncEnumerable([DynamicallyAccessedMembers(DynamicallyAcc
             }
         });
     }
+
+    public static bool TryGetStreamType(Type stream, [NotNullWhen(true)] out Type? streamType)
+        => _streamTypeLookup.TryGetValue(stream, out streamType);
```

Comment:
This seems fragile because it will give a false negative if we don't call `IsStreamingType(Type)` first. Can we insert null when the type is not a streaming type in, and assert that the key always exists here?